tests/testthat/test-epi_slide.R

```diff
@@ -1386,3 +1386,33 @@ test_that("`epi_slide_mean` errors when passed `col_names` as list", {
     class = "epiprocess__epi_slide_mean__col_names_in_list"
   )
 })
+
+test_that("epi_slide_mean produces same output as epi_slide_opt", {
+  result1 <- epi_slide_mean(small_x, value, before = 50, names_sep = NULL, na.rm = TRUE)
+  result2 <- epi_slide_opt(small_x, value,
+    f = data.table::frollmean,
+    before = 50, names_sep = NULL, na.rm = TRUE
+  )
+  expect_identical(result1, result2)
+
+  # # Theoretically should work, but getting "Error: C stack usage  7973092 is
+  # too close to the limit" (recursion is too deep)
+  # result3 <- epi_slide_opt(small_x, value, f = slider::slide_mean,
+  #   before = 50, names_sep = NULL, na_rm = TRUE)
```

Comment:
question: Not sure what to do about this. I do think the `slider` functions are useful to support, but this seems like an issue with their code.

`data.table` functions work fine, and I believe both it and `slider` are implemented in C++. [Other tests I've done](https://delphi-org.slack.com/archives/C028AUR6MV1/p1707760507375069) didn't have the recursion issue, so I suppose it is the result of `slider` inside of a `dplyr` call.

Reply:
I can reproduce this problem on my machine, though the actual stack size is different. Doesn't look like they have a corresponding error [here](https://github.com/r-lib/slider/issues)?

Reply:
actually, throwing a browser on the pile, looks like the problem is happening when trying to identify which function this is?

Reply:
[just sort of blindly following this advice](https://stackoverflow.com/questions/72219118/how-can-i-check-in-r-if-a-function-is-in-a-list-of-functions) seems to have worked. I'll add the suggestion above

Reply:
Oh, interesting. I was trying to use `all.equal` in case the user-passed function had been reexported (namespace wasn't `slider` or `data.table` even though the actual function behavior is the same). Odd that that causes problems. Thanks for finding a fix.